smarts/core/utils/math.py

```diff
@@ -336,11 +336,17 @@ def line_intersect(a, b, c, d) -> Union[np.ndarray, None]:
 
 
 def line_intersect_vectorized(
-    a: np.ndarray, b: np.ndarray, C: np.ndarray, D: np.ndarray
+    a: np.ndarray,
+    b: np.ndarray,
+    C: np.ndarray,
+    D: np.ndarray,
+    ignore_start_pt: bool = False,
 ) -> bool:
     """Vectorized version of `line_intersect(...)`, where C and D represent
     the segment points for an entire line, and a and b are points of a single
     line segment to be tested against.
+    If ignore_start_pt is True, then two diverging lines that *only* intersect at
```

Comment:
What was the need for this? It looks like this is only used when determining intersections in Opendrive.

Reply:
Yeah, it's for OpenDrive lanes that start at the same point but diverge from there; we don't want to count these as intersections (because the traffic in these two lanes does not interfere with each other).  This usually seems to happen\ when a single lane is being split into two.